### PR TITLE
update benchmarking labels

### PR DESCRIPTION
**Pull request by arnupretorius**
## What?
Update to ci for docker to reflect new benchmarking labels. Added new labels to be more specific about benchmarking status:

* Benchmark required - needs benchmarking, but has not yet been set to run.
* Benchmark in progress - experiments have been kicked off and are still running.
* Benchmarking is complete - results are in.

### CODE REVIEW

**Review by AsadJeewa — APPROVED** [review]
Thanks, @arnupretorius 
This will make reviewing a lot more streamlined :smile:

File: .github/workflows/push_docker.yaml [hunk 1]
```diff
@@ -103,7 +103,7 @@ jobs:
 
   # Action pushes new docker image if we have a PR to dev and have the label benchmark
   build-push-branches:
-    if:  ${{ github.event.label.name == 'benchmark' }}
+    if:  ${{ github.event.label.name == 'benchmark required' }}
```
Comment (AsadJeewa) [inline .github/workflows/push_docker.yaml:106]: One minor thing is that we could make it "benchmark complete" with a small 'b' so it is in line with other labels. Same for the maintenance label

Comment (arnupretorius) [reply .github/workflows/push_docker.yaml:106]: Nice catch, will do 👍

**Review by RuanJohn — APPROVED** [review]
Thank you @arnupretorius :)

**Pull request merged**